tests/scripts/thread-cert/node.py

```diff
@@ -3558,6 +3570,82 @@ def get_router_id_range(self):
         line = self._expect_command_output()[0]
         return [int(item) for item in line.split()]
 
+    def get_channel_monitor_info(self) -> Dict:
+        """
+        Returns:
+            Dict of channel monitor info, e.g. 
+                {'enabled': '1',
+                 'interval': '41000',
+                 'threshold': '-75',
+                 'window': '960',
+                 'count': '985',
+                 'occupancies': {
+                    '11': '0.00%',
+                    '12': '3.50%',
+                    '13': '9.89%',
+                    '14': '15.36%',
+                    '15': '20.02%',
+                    '16': '21.95%',
+                    '17': '32.71%',
+                    '18': '35.76%',
+                    '19': '37.97%',
+                    '20': '43.68%',
+                    '21': '48.95%',
+                    '22': '54.05%',
+                    '23': '58.65%',
+                    '24': '68.26%',
+                    '25': '66.73%',
+                    '26': '73.12%'
+                    }
+                }
+        """
+        config = {}
+        self.send_command('channel monitor')
+
+        for line in self._expect_results(r'\S+'):
+            try:
+                key, val = line.split(': ')
+                config.update({key: val})
+            except ValueError:
+                try:
+                    key, val = line.split(':')  # occupancy
+                    val = {}
+                    config.update({key: val})
+                except ValueError:
+                    # channel occupancies
+                    key = line.split()[1]
+                    val = line.split()[3]
+                    config['occupancies'].update({key: val})
```

Comment:
Maybe use python regex to do this? This is optional and up to you. It just seems weird to handle this in exceptions.